src/main/java/org/jabref/gui/groups/GroupDialog.java

```diff
@@ -304,39 +358,55 @@ public void actionPerformed(ActionEvent e) {
         searchGroupCaseSensitive.addItemListener(itemListener);
 
         // configure for current type
-        if ((editedGroup != null) && (editedGroup.getClass() == WordKeywordGroup.class)) {
-            WordKeywordGroup group = (WordKeywordGroup) editedGroup;
-            nameField.setText(group.getName());
-            keywordGroupSearchField.setText(group.getSearchField());
-            keywordGroupSearchTerm.setText(group.getSearchExpression());
-            keywordGroupCaseSensitive.setSelected(group.isCaseSensitive());
-            keywordGroupRegExp.setSelected(false);
-            keywordsRadioButton.setSelected(true);
-            setContext(editedGroup.getHierarchicalContext());
-        } else if ((editedGroup != null) && (editedGroup.getClass() == RegexKeywordGroup.class)) {
-            RegexKeywordGroup group = (RegexKeywordGroup) editedGroup;
-            nameField.setText(group.getName());
-            keywordGroupSearchField.setText(group.getSearchField());
-            keywordGroupSearchTerm.setText(group.getSearchExpression());
-            keywordGroupCaseSensitive.setSelected(group.isCaseSensitive());
-            keywordGroupRegExp.setSelected(true);
-            keywordsRadioButton.setSelected(true);
-            setContext(editedGroup.getHierarchicalContext());
-        } else if ((editedGroup != null) && (editedGroup.getClass() == SearchGroup.class)) {
-            SearchGroup group = (SearchGroup) editedGroup;
-            nameField.setText(group.getName());
-            searchGroupSearchExpression.setText(group.getSearchExpression());
-            searchGroupCaseSensitive.setSelected(group.isCaseSensitive());
-            searchGroupRegExp.setSelected(group.isRegularExpression());
-            searchRadioButton.setSelected(true);
-            setContext(editedGroup.getHierarchicalContext());
-        } else if ((editedGroup != null) && (editedGroup.getClass() == ExplicitGroup.class)) {
-            nameField.setText(editedGroup.getName());
-            explicitRadioButton.setSelected(true);
-            setContext(editedGroup.getHierarchicalContext());
-        } else { // creating new group -> defaults!
+        if (editedGroup == null) {
+            // creating new group -> defaults!
             explicitRadioButton.setSelected(true);
             setContext(GroupHierarchyType.INDEPENDENT);
+        } else {
+            if (editedGroup.getClass() == WordKeywordGroup.class) {
+                WordKeywordGroup group = (WordKeywordGroup) editedGroup;
+                nameField.setText(group.getName());
```

Comment:
According to the Java language specifications
> The type of the Expression must be char, byte, short, int, Character, Byte, Short, Integer, String, or an enum type, or a compile-time error occurs.

thus switch over a class type does not work.